src/pages/CommandView/CommandViewForm.tsx

```diff
@@ -0,0 +1,183 @@
+    setModel(handleByteParametersReset(initialData, model, command.parameters))
+  }
+
+  const onFormUpdated = (
+    changeEvent: IChangeEvent,
+    es: ErrorSchema | undefined,
+  ) => {
+    const formData = changeEvent.formData
+    setModel(formData)
+
+    const cleanedModel = cleanModelForDisplay(formData, command.parameters)
+    setDisplayModel(cleanedModel)
+  }
+
+  const raiseError = (message: string) => {
+    // TODO: make this more useful
+    console.error(message)
+  }
+
+  const handleError = (message: string) => {
+    // this is meant to do nothing but swallow the error message;
+    // the Form will automatically show validation messages in the UI and
+    // having this *do nothing* function prevents those messages from also
+    // being dumped to console.error
+  }
+
+  const onSubmit = () => {
+    if (hasByteParameters && isJob) {
+      raiseError('Bytes parameters are not supported for scheduled jobs')
+      return
+    }
+
+    const argumentToSubmit = getSubmitArgument(
+      model,
+      command,
+      isJob,
+      hasByteParameters,
+    )
+
+    const payload = prepareModelForSubmit(
+      argumentToSubmit,
+      command.parameters,
+      isJob,
+    )
+
+    const forwardPath = isJob ? '/jobs/' : '/requests/'
+
+    if (hasByteParameters) {
+      const data = new FormData()
+      const config = {
+        headers: { 'Content-type': 'multipart/form-data' },
+      } as AxiosRequestConfig<FormData>
+
+      data.append('request', JSON.stringify(payload))
+
+      for (const fileData of fileMetaData) {
+        const file = dataUrlToFile(fileData.dataUrl)
+        data.append(fileData.parameterName as string, file, file.name)
+      }
+
+      axiosInstance
+        .post('/api/v1/requests', data, config)
+        .then((response) => {
+          navigate(forwardPath + response.data.id)
+        })
+        .catch((error) => {
+          raiseError(error.toJSON())
+        })
+    } else {
+      const path = isJob ? '/api/v1/jobs' : '/api/v1/requests'
+
+      axiosInstance
+        .post(path, payload)
+        .then((response) => {
+          navigate(forwardPath + response.data.id)
+        })
+        .catch((error) => {
+          raiseError(error.toJSON())
+        })
+    }
+  }
+
+  const widgets = {
+    FileWidget: CustomFileWidget,
+  }
+
+  return (
+    <Box pt={2} display="flex" alignItems="flex-start">
+      <Box width={3 / 5}>
+        <BytesParameterContext.Provider
+          value={{ fileMetaData, setFileMetaData }}
+        >
+          <Form
+            schema={schema}
+            uiSchema={uiSchema}
+            formData={model}
+            onChange={onFormUpdated}
+            onSubmit={onSubmit}
+            onError={handleError}
+            validate={validator}
+            widgets={widgets}
+          >
+            <Button variant="contained" onClick={onResetForm} color="secondary">
+              Reset
+            </Button>
+            &nbsp;&nbsp;
```

Comment:
This is reflective of a problem I'd totally forgotten about. There seems to be a bug in the library, but I'm not sure. I've refactored this so that the (visible) buttons are no longer children of the form and it's *almost* right. But you can bring up *complex->echo_bool* to see that it's still got problems. I have no idea what's going on with this...